File: androidApp/src/main/kotlin/com/loop/mobile/presentation/home/HomeScreen.kt
package com.loop.mobile.presentation.home

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Button
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import androidx.navigation.NavController
import com.loop.mobile.domain.auth.AuthStateManager
import com.loop.mobile.presentation.navigation.Screen

@Composable
fun HomeScreen(navController: NavController, authStateManager: AuthStateManager) {
    val user by authStateManager.user.collectAsState()

    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(24.dp),
        verticalArrangement = Arrangement.Center,
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        if (user != null) {
            Text("Welcome, ${user!!.username}!",style = MaterialTheme.typography.bodyLarge)
        } else {
            Text("To create your own music space", style = MaterialTheme.typography.bodyLarge)
            Spacer(modifier = Modifier.height(16.dp))
            Button(onClick = { navController.navigate(Screen.Login.route) }) {
                Text("Login")
            }
        }
    }
}

File: iosApp/iosApp/Presentation/home/HomeScreen.swift
import SwiftUI
import shared

struct HomeScreen: View {
    @ObservedObject var authWrapper: AuthStateWrapper
    var onLoginTap: () -> Void

    var body: some View {
        VStack(spacing: 16) {
            if let user = authWrapper.user {
                Text("Welcome, \(user.username)!")
                    .font(.title)
            } else {
                Text("To create your own music space")
                    .font(.headline)

                Button("Login", action: onLoginTap)
                    .buttonStyle(.borderedProminent)
            }
        }
        .padding()
    }
}
